### PR TITLE
Use a vector for ItemRenderer::translate

### PR DESCRIPTION
**Pull request by tronical**
This simplifes call sites and implementations.

### CODE REVIEW

File: internal/core/items.rs [hunk 1]
```diff
@@ -983,11 +983,13 @@ impl Item for Rotate {
         backend: &mut ItemRendererRef,
         _self_rc: &ItemRc,
     ) -> RenderingResult {
-        let origin_x = self.logical_rotation_origin_x();
-        let origin_y = self.logical_rotation_origin_y();
-        (*backend).translate(origin_x, origin_y);
+        let origin = LogicalVector::from_lengths(
+            self.logical_rotation_origin_x(),
+            self.logical_rotation_origin_x(),
```
Comment (ogoffart) [inline internal/core/items.rs:988]: _y

File: internal/backends/winit/renderer/skia/itemrenderer.rs [hunk 2]
```diff
@@ -647,10 +647,10 @@ impl<'a> ItemRenderer for SkiaRenderer<'a> {
         from_skia_rect(&self.canvas.local_clip_bounds().unwrap_or_default()) / self.scale_factor
     }
 
-    fn translate(&mut self, x: LogicalLength, y: LogicalLength) {
+    fn translate(&mut self, distance: LogicalVector) {
         self.canvas.translate(skia_safe::Vector::from((
-            (x * self.scale_factor).get(),
-            (y * self.scale_factor).get(),
+            (distance.x_length() * self.scale_factor).get(),
```
Comment (ogoffart) [inline internal/backends/winit/renderer/skia/itemrenderer.rs:652]: You can multiply the vector by the scale factor then access the X and y directly

Comment (tronical) [reply internal/backends/winit/renderer/skia/itemrenderer.rs:652]: Good idea

File: internal/core/items.rs (hunk 1 above)
Comment (tronical) [reply internal/core/items.rs:988]: Oops!

File: internal/backends/winit/renderer/femtovg/itemrenderer.rs [hunk 3]
```diff
@@ -882,12 +882,13 @@ impl<'a> ItemRenderer for GLItemRenderer<'a> {
         None
     }
 
-    fn translate(&mut self, x: LogicalLength, y: LogicalLength) {
-        self.canvas
-            .borrow_mut()
-            .translate((x * self.scale_factor).get(), (y * self.scale_factor).get());
+    fn translate(&mut self, distance: LogicalVector) {
+        self.canvas.borrow_mut().translate(
+            (distance.x_length() * self.scale_factor).get(),
```
Comment (ogoffart) [inline internal/backends/winit/renderer/femtovg/itemrenderer.rs:887]: Here too

**Review by ogoffart — APPROVED** [review]

**Pull request merged**